typescript/sdk/src/providers/transactions/submitter/ethersV5/EV5GnosisSafeTxBuilderWriter.ts

```diff
@@ -0,0 +1,77 @@
+import { SafeTransactionData } from '@safe-global/safe-core-sdk-types';
+
+import { assert } from '@hyperlane-xyz/utils';
+
+// prettier-ignore
+// @ts-ignore
+import { getSafe, getSafeService } from '../../../../utils/gnosisSafe.js';
+import { MultiProvider } from '../../../MultiProvider.js';
+import { GnosisTransactionBuilderPayload } from '../../../ProviderType.js';
+import { PopulatedTransaction, PopulatedTransactions } from '../../types.js';
+import { TxSubmitterType } from '../TxSubmitterTypes.js';
+
+import { EV5GnosisSafeTxSubmitter } from './EV5GnosisSafeTxSubmitter.js';
+import { EV5GnosisSafeTxBuilderProps } from './types.js';
+
+/**
+ * This class is used to create a Safe Transaction Builder compatible object.
+ * It is not a true Submitter because it does not submits any transactions.
+ */
+export class EV5GnosisSafeTxBuilderFactory extends EV5GnosisSafeTxSubmitter {
+  public readonly txSubmitterType: TxSubmitterType =
+    TxSubmitterType.GNOSIS_TX_BUILDER;
+  constructor(
+    public readonly multiProvider: MultiProvider,
+    public readonly props: EV5GnosisSafeTxBuilderProps,
+    safe: any,
+    safeService: any,
+  ) {
+    super(multiProvider, props, safe, safeService);
+  }
+
+  static async create(
+    multiProvider: MultiProvider,
+    props: EV5GnosisSafeTxBuilderProps,
+  ): Promise<EV5GnosisSafeTxBuilderFactory> {
+    const { chain, safeAddress } = props;
+    const { gnosisSafeTransactionServiceUrl } =
+      multiProvider.getChainMetadata(chain);
+    assert(
+      gnosisSafeTransactionServiceUrl,
+      `Must set gnosisSafeTransactionServiceUrl in the Registry metadata for ${chain}`,
+    );
+    const safe = await getSafe(chain, multiProvider, safeAddress);
+    const safeService = await getSafeService(chain, multiProvider);
+
+    return new EV5GnosisSafeTxBuilderFactory(
+      multiProvider,
+      props,
+      safe,
+      safeService,
+    );
+  }
```

Comment:
this seems pretty identical to the other factory

Reply:
yeah this is a duplicate file. Removing